ssl-bootstrap.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+#Script to create and sign broker certificate for SSL operation. In default mode will
+#generate generic default values for DNs and passwords.
+
+#The following environment should be supplied to Kafka to support this default behavior
+export KAFKA_SSL_KEYSTORE_LOCATION=${KAFKA_SSL_KEYSTORE_LOCATION-/etc/ssl/private/server.keystore.jks}
+export KAFKA_SSL_KEYSTORE_PASSWORD=${KAFKA_SSL_KEYSTORE_PASSWORD-password1234}
+export KAFKA_SSL_KEY_PASSWORD=${KAFKA_SSL_KEY_PASSWORD-password1234}
+export KAFKA_SSL_TRUSTSTORE_LOCATION=${KAFKA_SSL_TRUSTSTORE_LOCATION-/etc/ssl/private/server.truststore.jks}
+export KAFKA_SSL_TRUSTSTORE_PASSWORD=${KAFKA_SSL_TRUSTSTORE_PASSWORD-password1234}
+
+#The following environment is optional
+#export KAFKA_SSL_CLIENT_AUTH=none
+#export KAFKA_SSL_KEYSTORE_TYPE=JKS
+#export KAFKA_SSL_TRUSTSTORE_TYPE=JKS
+export KAFKA_SSL_SECURE_RANDOM_IMPLEMENTATION=${KAFKA_SSL_SECURE_RANDOM_IMPLEMENTATION-SHA1PRNG}
+
+#The following environment is not used by Kafka but controls the creation of bootstrap certificates
+# CA_DN for the DN used to generate the Root CA (in a format supported by openssl)
+# CA_PASSWORD to protect the CA private key
+# CERT_DN for the DN used to generate the server private key (in a format supported by keytool)
+export CA_KEY=${CA_KEY-/etc/ssl/private/ca_key} #The CA Private key used for signing
+export CA_PASSWORD=${CA_PASSWORD-password1234} #The CA Private key password
+export CA_P12_FILE=${CA_P12_FILE-} #Use an outside CA key pair as a PKCS12 file
+export CA_P12_PASSWORD=${CA_PA12_PASSWORD-password1234} #the password for the P12 file
+
+cd /etc/ssl/private
+
+#Check if the Keystore and/or Truststore files are available as Secrets; if they are, then
+#this script does not need to run.
+if [[ -f /run/secrets/server.keystore.jks ]]; then
```

Comment:
It might be better to make it user-explicit which keystore and truststore they are using, rather than automatically setting it. Especially as I think this is just for swarm secrets